### PR TITLE
Add bufferEntireResponse flag where needed

### LINKED ISSUE #262: Add bufferEntireResponse=true flag to reporter config requests, where appropriate
_Depends on https://github.com/VEuPathDB/WDK/pull/53_

The backend will be updated to accept a queryParam flag `bufferEntireResponse` to not stream the response. The client will set that flag anywhere a reporter response is used by the client application (as opposed to being streamed to the browser for download purposes).

### PR DESCRIPTION
**Pull request by adnauseum**
**Before:**

Screen Shot 2023-01-23 at 1 19 43 PM

**After**:

image

Companion PR: https://github.com/VEuPathDB/web-app-preferred-organisms/pull/9

### CODE REVIEW

File: Client/src/Service/Mixins/SearchReportsService.ts [hunk 1]
```diff
@@ -68,8 +68,15 @@ export default (base: ServiceBase) => {
     const question = await base.findQuestion(answerSpec.searchName);
     const recordClass = await base.findRecordClass(question.outputRecordClassName);
     let url = base.getStandardSearchReportEndpoint(recordClass.urlSegment, question.urlSegment);
-    let searchConfig: SearchConfig = answerSpec.searchConfig
-    let body: StandardSearchReportRequest = { searchConfig, reportConfig };
+    let searchConfig: SearchConfig = answerSpec.searchConfig;
+    const reportConfigWithResponseBufferingSet: StandardReportConfig = {
+      ...reportConfig,
+      bufferEntireResponse: reportConfig.bufferEntireResponse ?? false,
```
Comment (dmfalke) [inline Client/src/Service/Mixins/SearchReportsService.ts:74]: I was thinking that this should default to `true`. What do you think?

**Review by dmfalke — COMMENTED** [review]
I think we want to apply it to any method making a fetch request to the standard json reporter endpoint. The only place where we definitely want it to be false is in the `downloadAnswer` method. Does this sound right to you?

File: Client/src/Service/Mixins/SearchReportsService.ts (hunk 1 above)
Comment (adnauseum) [reply Client/src/Service/Mixins/SearchReportsService.ts:74]: Ah, I see. I got terribly confused. That makes more sense now. I will amend this PR!

Comment (adnauseum) [reply Client/src/Service/Mixins/SearchReportsService.ts]: > I think we want to apply it to any method making a fetch request to the standard json reporter endpoint. The only place where we definitely want it to be false is in the `downloadAnswer` method. Does this sound right to you?

So would this mean making a PR `web-app-preferred-organisms` to add the `bufferEntireResponse` flag here: https://github.com/VEuPathDB/web-app-preferred-organisms/blob/main/src/lib/utils/organismMetadata.ts#L43 or here: https://github.com/VEuPathDB/web-app-preferred-organisms/blob/main/src/lib/utils/datasetMetadata.ts#L56?

**Review by dmfalke — APPROVED** [review]
Looks great! I confirmed that results are buffered in the strategy workspace, but not when downloading.

**Pull request merged**